Source/RequestTaskMap.swift

```diff
@@ -28,14 +28,18 @@ import Foundation
 struct RequestTaskMap {
     private var tasksToRequests: [URLSessionTask: Request]
     private var requestsToTasks: [Request: URLSessionTask]
+    private var taskEvents: [URLSessionTask: (completed: Bool, metricsGathered: Bool)]
```

Comment:
I think I'd recommend making a `TaskEvent` struct with `completed` and `metricsGathered` properties to make the switches easier to follow. Right now you're switching on `true` and `false` values without enough context.

Reply:
Ah, that would make the calculations easier.

Reply:
Actually, looking into it, it doesn't really make the calculations any easier. What exactly did you have in mind? AFAICT, I'll need to switch on the booleans either way.

Reply:
It would just make the switch more clear:

```swift
switch events {
case (true, _): fatalError("RequestTaskMap consistency error: duplicate completionReceivedForTask call.")
case (false, false): taskEvents[task] = (completed: true, metricsGathered: false)
case (false, true): self[task] = nil
}
```

vs.

```swift
switch (events.completed, events.metricsGathered) {
case (true, _): fatalError("RequestTaskMap consistency error: duplicate completionReceivedForTask call.")
case (false, false): taskEvents[task] = (completed: true, metricsGathered: false)
case (false, true): self[task] = nil
}
```

Right now, I have no idea what's on the left and what's on the right when looking at the switch.

Reply:
Okay, I can do that.